File: backend/internal/contract/policy.go
package contract

// ============================================================================
// [分工1] 控制平面 → 执行平面：三门统一策略引擎
// ============================================================================

// PolicyEngine [UNDEFINED] [分工1] 三门统一策略引擎接口
//
// 控制平面暴露给执行平面的能力：评分、决策、策略配置。
// 当前实现状态：完全未定义，三门各自独立判断
type PolicyEngine interface {
	Score(eventContext interface{}) (int, []string)
	ShouldBlock(riskScore int) bool
	ShouldDegrade(riskScore int) bool
	ShouldHumanReview(riskScore int) bool
}

// PolicyRule [分工3] 策略规则定义
type PolicyRule struct {
	ID        string `json:"id"`
	Name      string `json:"name"`
	GateType  string `json:"gate_type"`
	Enabled   bool   `json:"enabled"`
	Priority  int    `json:"priority"`
	Condition string `json:"condition"`
	Action    string `json:"action"`
}

// PolicyConfig [分工3] 策略配置
type PolicyConfig struct {
	Mode  string       `json:"mode"`
	Rules []PolicyRule `json:"rules"`
}

File: frontend/src/api/policy.ts
import { http } from "@/utils/http";

export type PolicyRule = {
  id: string;
  name: string;
  description: string;
  gate_type: "message" | "action" | "return";
  condition: string;
  action: "Allow" | "Block" | "Degrade" | "Deny" | "HumanApproval";
  priority: number;
  enabled: boolean;
  risk_threshold: number;
};

export type CreateRulePayload = {
  name: string;
  description: string;
  gate_type: "message" | "action" | "return";
  condition: string;
  action: "Allow" | "Block" | "Degrade" | "Deny" | "HumanApproval";
  priority: number;
  enabled: boolean;
  risk_threshold: number;
};

export type PolicyConfig = {
  risk_weights: {
    alpha: number;
    beta: number;
    gamma: number;
  };
  global_threshold: number;
  rules: PolicyRule[];
};

type ApiResult<T> = { code: number; data: T; message?: string };

export const getPolicyConfig = () => {
  return http.request<ApiResult<PolicyConfig>>("get", "/aegis/policy/config");
};

export const getPolicyRules = (params?: object) => {
  return http.request<ApiResult<PolicyRule[]>>("get", "/aegis/policy/rules", {
    params
  });
};

export const createPolicyRule = (data: CreateRulePayload) => {
  return http.request<ApiResult<PolicyRule>>("post", "/aegis/policy/rules", {
    data
  });
};

export const updatePolicyRule = (data: PolicyRule) => {
  return http.request<ApiResult<PolicyRule>>("put", "/aegis/policy/rules", {
    data
  });
};

export const deletePolicyRule = (id: string) => {
  return http.request<ApiResult<null>>("delete", `/aegis/policy/rules/${id}`);
};

export const reorderPolicyRules = (ruleIds: string[]) => {
  return http.request<ApiResult<null>>("put", "/aegis/policy/rules/reorder", {
    data: { rule_ids: ruleIds }
  });
};

export const updatePolicyConfig = (data: Partial<PolicyConfig>) => {
  return http.request<ApiResult<null>>("put", "/aegis/policy/config", {
    data
  });
};
